File: 2107-find-unique-binary-string/find-unique-binary-string.go
func findDifferentBinaryString(nums []string) string {
    arr := []string{"0","1"}

    n := len(nums)
    if n == 1 {
        if nums[0] == "0" {
            return "1"
        } else {
            return "0"
        }
        
    }
    mp := make(map[string]bool)

    for _, k := range nums {
        mp[k] = true
    }

    return calc(arr, n-1, arr, mp)
}

func calc(arr []string, n int, prev []string, mp map[string]bool) string {
    if n == 1 {
        for _, k := range prev {
            for _, k1 := range arr {
                _, ok := mp[k + k1]
                if !ok {
                    return k + k1
                }
            }
            
        }
    }
    t := make([]string, 0)
    for _, k1 := range arr {
        for _, k := range prev {
            t = append(t, k1 + k)
        }
    }
    return calc(arr, n-1 , t, mp)

}

File: 2107-find-unique-binary-string/find-unique-binary-string.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        arr = []
        s = set()
        v = 0
        for k in nums:
            arr.append(int(k, 2))
            s.add(int(k, 2))
        #print(arr)
        for k in range(0, pow(2,len(nums[0]))):
            if k not in s:
                v = k
                break
        x = format(v, 'b')
        #print(x)
        if len(nums[0]) > len(str(x)):
            return ("0" * (len(nums[0]) - len(str(x)))) + str(x)
        return str(x)
        


